src/Tools/dotnet-monitor/Egress/S3/MultiPartUploadStream.cs

```diff
@@ -120,7 +167,18 @@ private async Task DoWriteAsync(bool allowPartialWrite, CancellationToken cancel
 
     public override void Write(byte[] buffer, int offset, int count)
     {
-        throw new NotSupportedException();
+        if (Closed)
```

Comment:
Please move this implementation to an override of the `Write(ReadOnlySpan<byte>)` method and have this Write method call that one: `Write(buffer.AsMemory().Slice(offset, count));`